### DISCUSSION TITLE
The development guidelines don't explain what to do after making a local clone of the formula

### DISCUSSION THREAD

**Discussion by tzz** [d12frosted/homebrew-emacs-plus#715] [General]
It would be helpful if the "Building locally" development docs said what to do after `build ...` for those who don't know Homebrew wel but want to debug issues in emacs-plus:

* how to modify the formula once it's cloned
* where to find the source files
* how to modify the source files, test the build with the changes, and generate a patch with the changes

**Comment by d12frosted** [comment]
Hey @tzz ,

Small clarification. By 'source files' what do you mean? Emacs sources or the formula sources?

**Comment by tzz** [comment]
In #551 I needed the Emacs source files, but I think I may also need to modify the formula source files (patches and build commands).

**Comment by d12frosted** [comment]
I see. Not sure how that would help you, though. 

I am totally in favor of improving the documentation, but I want to maintain only documentation relevant to the Emacs+ formula (as it's not the simplest formula out in the homebrew world). I definitely don't want to maintain a documentation explaining how to debug builds with `brew` (their documentation explains that well) and definitely don't want to explain how to develop Emacs (there is documentation for that). Same goes with using `git`. That being said, Emacs+ lacks better documentation of it's own parts. And maybe it can link mentioned docs to make things simpler for new to be contributors.

Also, I want to stress out two important things. 

1. Emacs+ is just a formula of GNU Emacs for homebrew. It's not a distribution of Emacs (e.g. `emacs-mac` formula).
2. I rarely accept patches to Emacs sources. There are two cases when I do accept them. (1) When there is a critical patch that should be merged soon into upstream, but I want to stop the 'bleeding' ASAP. (2) When there is a macos-only feature that will be unlikely accepted into upstream, but Emacs+ users find it useful. In the first, the patch is usually somewhere in the mailing list. In the second case explaining how to use `build` script IMO should be enough. Everything else (e.g. developing Emacs) should be found in Emacs documentation.

**Comment by tzz** [comment]
ACK on both. Thank you for the explanations.

It's still unclear to me how to access the build directory when it fails, unfortunately. If that's a `brew` question then I'll head that way, but it seems this formula does some special setup.

If it's possible to add some links to the development docs here to `brew` or other docs answering my questions, that would probably help quite a few casual contributors.
